File: server/models/redirect.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class Redirect(BaseModel):
    id: Optional[str] = Field(None, alias="_id")
    from_url: str
    to_url: str
    is_active: bool = True
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        allow_population_by_field_name = True
        schema_extra = {
            "example": {
                "from_url": "/old-product-name",
                "to_url": "/new-product-name",
                "is_active": True
            }
        }

File: server/models/Redirect.js
const mongoose = require('mongoose');

const redirectSchema = new mongoose.Schema(
  {
    fromUrl: {
      type: String,
      required: true,
      lowercase: true,
      trim: true,
      unique: true,
      help: 'Original broken or old URL path (e.g., /old-product-name)'
    },
    toUrl: {
      type: String,
      required: true,
      lowercase: true,
      trim: true,
      help: 'New destination URL path (e.g., /new-product-name)'
    },
    isActive: {
      type: Boolean,
      default: true,
      help: 'Whether this redirect is currently active'
    },
    createdAt: {
      type: Date,
      default: Date.now
    },
    updatedAt: {
      type: Date,
      default: Date.now
    }
  },
  { timestamps: true }
);

// Index for faster lookups
redirectSchema.index({ isActive: 1 });

module.exports = mongoose.model('Redirect', redirectSchema);
